engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -114,4 +134,309 @@ protected void destroy() {
         super.destroy();
         UpdateGraphProcessor.DEFAULT.removeSource(this);
     }
+
+    private static class SyntheticDateTimeSource implements
+            ColumnSource<DateTime>,
+            DefaultChunkSource.WithPrev<Values>,
+            MutableColumnSourceGetDefaults.LongBacked<DateTime>,
+            FillUnordered,
+            InMemoryColumnSource {
+
+        private DateTime baseTime;
+        private final long period;
+
+        private SyntheticDateTimeSource(long period) {
+            this.period = period;
+        }
+
+        @Override
+        public DateTime get(long index) {
+            return DateTimeUtils.plus(baseTime, period * index);
+        }
+
+        @Override
+        public long getLong(long index) {
+            return baseTime.getNanos() + period * index;
+        }
+
+        @Override
+        public DateTime getPrev(long index) {
+            return get(index);
+        }
+
+        @Override
+        public long getPrevLong(long index) {
+            return getLong(index);
+        }
+
+        @Override
+        public void fillChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Values> destination,
+                @NotNull RowSequence rowSequence) {
+            final WritableObjectChunk<DateTime, ? super Values> objectDest = destination.asWritableObjectChunk();
+            objectDest.setSize(0);
+            rowSequence.forAllRowKeys(ii -> objectDest.add(DateTimeUtils.plus(baseTime, period * ii)));
+        }
+
+        @Override
+        public void fillPrevChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Values> destination,
+                @NotNull RowSequence rowSequence) {
+            fillChunk(context, destination, rowSequence);
+        }
+
+        @Override
+        public Class<DateTime> getType() {
+            return DateTime.class;
+        }
+
+        @Override
+        public Class<?> getComponentType() {
+            return null;
+        }
+
+        @Override
+        public WritableRowSet match(boolean invertMatch, boolean usePrev, boolean caseInsensitive, RowSet selection,
```

Comment:
This might be the only override of `match` in the entire code base. Everything else extends `AbstractColumnSource`. I'll need to read this more carefully tomorrow.

Reply:
I'm convinced that what you have is correct.